src/vm.rs

```diff
@@ -176,29 +176,41 @@ impl Vm {
             }
         }
 
-        info!("args: {:?}", args);
-        let cmd = cmd.args(args);
-        let vm_process: Child = cmd
-            .stdin(Stdio::null())
-            .stderr(Stdio::piped())
-            .spawn()
-            .map_err(|e| Error::ProcessSpawn(e))?;
-        ts_vec.push(Instant::now());
-
-        if force_exit {
-            let output = vm_process
-                .wait_with_output()
-                .expect("failed to wait on child");
-            let mut status = 0;
-            if !output.status.success() {
-                eprintln!("{:?}", String::from_utf8_lossy(&output.stderr));
-                status = 1;
+        let runtime = tokio::runtime::Builder::new_current_thread().enable_io().build().unwrap();
+        let (conn, vm_process) = runtime.block_on(async {
+            info!("args: {:?}", args);
+            let mut vm_process = Command::new(firerunner).args(args).kill_on_drop(true)
+                .stdin(Stdio::null())
+                .stderr(Stdio::piped())
+                .spawn()
+                .map_err(|e| Error::ProcessSpawn(e))?;
+            ts_vec.push(Instant::now());
+
+
+            if force_exit {
+                let output = vm_process .wait_with_output().await
+                    .expect("failed to wait on child");
+                let mut status = 0;
+                if !output.status.success() {
+                    eprintln!("{:?}", String::from_utf8_lossy(&output.stderr));
+                    status = 1;
+                }
+                crate::unlink_unix_sockets();
+                std::process::exit(status);
             }
-            crate::unlink_unix_sockets();
-            std::process::exit(status);
-        }
 
-        let (conn, _) = vm_listener.accept().unwrap();
+            let vm_listener = tokio::net::UnixListener::from_std(vm_listener).expect("convert from UnixListener std");
+            let conn = tokio::select! {
```

Comment:
It seems that this connection is by default in non-blocking mode.
I tested with
```shell
target/debug/fc_wrapper --kernel resources/images/vmlinux-4.20.0 --mem_size 128 --rootfs hello.ext4 --vcpu_count 1 < resources/hello.json
``` 
The `read_exact` at line `src/vm.rs:310` always failed with
```shell
VsockRead(Os { code: 11, kind: WouldBlock, message: "Resource temporarily unavailable" })
```